Fix command with complete but no nargs (#724)

gen4438 opened the pull request:
This PR fixes #724.

rtmor commented on vim/vscode-file-commands.vim:
```diff
@@ -31,7 +31,7 @@ command! -bang -nargs=? Ex call <SID>editOrNew(<q-args>, <q-bang>)
 command! -bang Find call VSCodeNotify('workbench.action.quickOpen')
 
-command! -complete=file -bang Write if <q-bang> ==# '!' | call VSCodeNotify('workbench.action.files.saveAs') | else | call VSCodeNotify('workbench.action.files.save') | endif
+command! -bang Write if <q-bang> ==# '!' | call VSCodeNotify('workbench.action.files.saveAs') | else | call VSCodeNotify('workbench.action.files.save') | endif
```
This appears to work, however, are there any advantages to removing the `-complete=file` parameter over simply including the `nargs=?` parameter? Both appear to fix the issue.

```suggestion
command! -complete=file -bang -nargs=? Write if <q-bang> ==# '!' | call VSCodeNotify('workbench.action.files.saveAs') | else | call VSCodeNotify('workbench.action.files.save') | endif
```

rtmor commented on vim/vscode-tab-commands.vim:
```diff
@@ -7,7 +7,7 @@ function! s:switchEditor(...) abort
 
 command! -complete=file -nargs=? Tabedit if empty(<q-args>) | call VSCodeNotify('workbench.action.quickOpen') | else | call VSCodeExtensionNotify('open-file', expand(<q-args>), 0) | endif
-command! -complete=file Tabnew call VSCodeExtensionNotify('open-file', '__vscode_new__', 0)
+command! Tabnew call VSCodeExtensionNotify('open-file', '__vscode_new__', 0)
```
```suggestion
command! -complete=file -nargs=? Tabnew call VSCodeExtensionNotify('open-file', '__vscode_new__', 0)
```

rtmor reviewed (commented):
These changes appear to work, however, is there an advantage to removing the `-complete=file` parameter entirely, as opposed to appending `nargs=?` to each command?

gen4438 commented:
Thank you for your review! Your suggestion looks better than my changes, so I've incorporated them.

(Actually, I modified it as you suggested at first, but since the default value of `: command-nargs` of nvim is` -nargs = 0` and does not accept arguments, I committed it so that the behavior does not change.)

mcaple commented:
Would be really nice if someone could approve this

GerardKetuma commented:
> Would be really nice if someone could approve this

Yep, also waiting on this to be approved. My :w does not work and also getting same E1208 output messages.

Kilill commented:
Tested this, seems to work fine, would someone please commit it ?, so that i don't have to patch it every time reinstall vscode :-)

mcaple commented:
Sorry @asvetliakov but any chance of reviewing this?

jsonMartin commented:
Also getting this error, looking forward to a merge of this 👍

francois-pasquier commented:
is there some update on this?

Kilill reviewed (approved):
Aplied this and has been working without a hitch

Kilill commented:
For the impatient , I applied the patch manually by directly editing the files works quite ok ( you find the files in your <Home Directory>/.vscode/extenstions/asvetliakov.vscode-neovim-0.0.82/vim

asvetliakov commented:
Thank you for the fix!

willopez commented:
@asvetliakov can we get a new version publish for this update? that would be great, thanks!

asvetliakov commented:
Published v0.0.83
